packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -495,6 +496,42 @@ class _AnimatedPositionedState extends AnimatedWidgetBaseState<AnimatedPositione
   }
 }
 
+/// Animated widget that automatically transitions the child's size over a given
+/// duration whenever the given child's size changes.
+class AnimatedSize extends SingleChildRenderObjectWidget {
+  /// Creates a widget that animates the size of its child implicitly.
+  ///
+  /// The [curve] and [duration] arguments must not be null.
+  AnimatedSize({
+    Key key,
+    Widget child,
+    this.curve: Curves.linear,
+    @required this.duration
+  }) : super(key: key, child: child);
+
+  /// The curve to apply when animating the parameters of this container.
+  final Curve curve;
+
+  /// The duration over which to animate the parameters of this container.
+  final Duration duration;
+
+  @override
+  RenderAnimatedSize createRenderObject(BuildContext context) {
+    return new RenderAnimatedSize(
+      duration: duration,
+      curve: curve
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, RenderObject renderObject) {
+    RenderAnimatedSize renderAnimatedSize = renderObject;
```

Comment:
You can just put this type in the function parameter list and avoid having this local variable.